app/javascript/pages/SignIn.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react"
+
+import { SignInForm } from "../authentication/SignInForm"
+import { LoggedInUserIdleTimeout } from "../authentication/timeout"
+import { UserProvider } from "../authentication/UserContext"
+import FormsLayout from "../layouts/Forms"
+
+const SignIn = () => {
+  return (
+    <FormsLayout>
+      <UserProvider>
```

Comment:
i think we should have a component wrapper function that applies all of the application-y stuff on top of each page.

```
function withAppSetup<T>(Component: React.ComponentType<T>) {
  return (props: T) => (
      <UserProvider>
        <LoggedInUserIdleTimeout />
        <Component {...props} />
      </UserProvider>
}
```

and then in SignIn / Homepage you can do 

`export withAppSetup(SignIn)`